lib/Service/ObjectHandlers/SaveObject.php

```diff
@@ -1250,35 +1253,852 @@ public function setDefaults(ObjectEntity $objectEntity): ObjectEntity
+            if (is_string($fileInput)) {
+                // Handle string inputs (base64, data URI, or URL)
+                return $this->processStringFileInput($objectEntity, $fileInput, $propertyName, $fileConfig, $index);
+            } elseif (is_array($fileInput) && $this->isFileObject($fileInput)) {
+                // Handle file object input
+                return $this->processFileObjectInput($objectEntity, $fileInput, $propertyName, $fileConfig, $index);
+            } else {
+                throw new Exception("Unsupported file input type for property '$propertyName'");
+            }
+        } catch (Exception $e) {
+            error_log("Error processing file property '$propertyName': " . $e->getMessage());
+            throw $e;
+        }
+
+    }//end processSingleFileProperty()
+
+
+    /**
+     * Processes string file input (base64, data URI, or URL).
+     *
+     * @param ObjectEntity $objectEntity The object entity being saved.
+     * @param string       $fileInput    The string input (base64, data URI, or URL).
+     * @param string       $propertyName The name of the file property.
+     * @param array        $fileConfig   The file property configuration from schema.
+     * @param int|null     $index        Optional index for array properties.
+     *
+     * @return int The ID of the created file.
+     *
+     * @throws Exception If file processing fails.
+     *
+     * @psalm-param ObjectEntity $objectEntity
+     * @phpstan-param ObjectEntity $objectEntity
+     * @psalm-param string $fileInput
+     * @phpstan-param string $fileInput
+     * @psalm-param string $propertyName
+     * @phpstan-param string $propertyName
+     * @psalm-param array<string, mixed> $fileConfig
+     * @phpstan-param array<string, mixed> $fileConfig
+     * @psalm-param int|null $index
+     * @phpstan-param int|null $index
+     * @psalm-return int
+     * @phpstan-return int
+     */
+    private function processStringFileInput(
+        ObjectEntity $objectEntity,
+        string $fileInput,
+        string $propertyName,
+        array $fileConfig,
+        ?int $index = null
+    ): int {
+        // Check if it's a URL
+        if (filter_var($fileInput, FILTER_VALIDATE_URL) && 
+            (strpos($fileInput, 'http://') === 0 || strpos($fileInput, 'https://') === 0)) {
+            // Fetch file content from URL
+            $fileContent = $this->fetchFileFromUrl($fileInput);
+            $fileData = $this->parseFileDataFromUrl($fileInput, $fileContent);
+        } else {
+            // Parse as base64 or data URI
+            $fileData = $this->parseFileData($fileInput);
+        }
+        
+        // Validate file against property configuration
+        $this->validateFileAgainstConfig($fileData, $fileConfig, $propertyName, $index);
+        
+        // Generate filename
+        $filename = $this->generateFileName($propertyName, $fileData['extension'], $index);
```

Comment:
Maybe good to check if we know the original filename before (you can for example derive it from the content-disposition headers when fetching a file from url),